internal/sdk/resource_encode.go

```diff
@@ -71,11 +71,18 @@ func recurse(objType reflect.Type, objVal reflect.Value, debugLogger Logger) (ou
 				debugLogger.Infof("Setting %q to %d", structTags.hclPath, iv)
 				output[structTags.hclPath] = iv
 
-			case reflect.Float32, reflect.Float64:
+			case reflect.Float64:
 				fv := fieldVal.Float()
 				debugLogger.Infof("Setting %q to %f", structTags.hclPath, fv)
 				output[structTags.hclPath] = fv
 
+			// (@jackofallops) - float32 support is here for completeness, but it will likely result in either rounding errors or type problems due to expectation of float64 inside Terraform
+			// Should be removed?
```

Comment:
yeah since we should be using `int64` everywhere going forwards, I suspect it'd be good to remove this (and add a unit test to check the Typed Models don't contain `int`, `int32` or `float32`?